docs/source/onedal/algorithms/ensembles/decision-forest.rst

```diff
@@ -0,0 +1,218 @@
+.. * distributed under the License is distributed on an "AS IS" BASIS,
+.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. * See the License for the specific language governing permissions and
+.. * limitations under the License.
+.. *******************************************************************************/
+
+.. default-domain:: cpp
+
+.. _alg_df:
+
+===================================
+Decision Forest Classification and Regression (DF)
+===================================
+
+Decision Forest (DF) :capterm:`classification` and :capterm:`regression` algorithms are based on an ensemble of 
+tree-structured classifiers, which are known as decision trees. Decision forest is built 
+using the general technique of bagging, a bootstrap aggregation, and a random choice of features. Decision tree is
+a binary tree graph. Its internal (split) nodes represent a decision function used to select the child node at the 
+prediction stage. Its leaf, or terminal, nodes represent the corresponding response values, which are the result 
+of the prediction from the tree. For more details, see [Breiman84]_ and [Breiman2001]_.
+
+.. |t_math| replace:: `Training <df_t_math_>`_
+.. |t_dense| replace:: `Dense <df_t_math_dense_>`_
+.. |t_hist| replace:: `Hist <df_t_math_hist_>`_
+.. |t_input| replace:: `train_input <df_t_api_input_>`_
+.. |t_result| replace:: `train_result <df_t_api_result_>`_
+.. |t_op| replace:: `train(...) <df_t_api_>`_
+
+.. |i_math| replace:: `Inference <df_i_math_>`_
+.. |i_dense| replace:: `Dense <df_i_math_dense_>`_
+.. |i_hist| replace:: `Hist <df_i_math_hist_>`_
+.. |i_input| replace:: `infer_input <df_i_api_input_>`_
+.. |i_result| replace:: `infer_result <df_i_api_result_>`_
+.. |i_op| replace:: `infer(...) <df_i_api_>`_
+
+=============== ============= ============= ======== =========== ============
+ **Operation**  **Computational methods**     **Programming Interface**
+--------------- --------------------------- ---------------------------------
+   |t_math|        |t_dense|     |t_hist|    |t_op|   |t_input|   |t_result|
+   |i_math|        |i_dense|     |i_hist|    |i_op|   |i_input|   |i_result|
+=============== ============= ============= ======== =========== ============
+
+------------------------
+Mathematical formulation
+------------------------
+
+.. _df_t_math:
+
+Training
+--------
+Given :math:`n` feature vectors :math:`X=\{x_1=(x_{11},\ldots,x_{1p}),\ldots,x_n=(x_{n1},\ldots,x_{np})\}` of
+size :math:`p`, their non-negative observation weights :math:`W=\{w_1,\ldots,w_n\}` and :math:`n` responses :math:`Y=\{y_1,\ldots,y_n\}` (for classification, :math:`y_i \in \{0, \ldots, \mathrm{C-1}\}`, where :math:`C` is the number of classes), the problem is to build a decision forest classification or regression model.
+
+During the training stage, :math:`\mathrm{M}` independent classification or regression trees are created using the following:
```

Comment:
```suggestion
During the training stage, :math:`M` independent classification or regression trees are created using the following:
```

`\mathrm{...}` is recommended when you want to use multiple letters. For example `I_{\mathrm{reduction}}`. Without mathrm it looks like multiplication of `r e d u ...`.

Do not use `\mathrm{...}` for one letter variables.